cpp/tests/orderby/is_sorted/is_sorted_tests.cu

```diff
@@ -0,0 +1,335 @@
+
+    auto actual = cudf::experimental::is_sorted(in, order, null_precedence);
+
+    EXPECT_EQ(false, actual);
+}
+
+TYPED_TEST(IsSortedNumeric, NullsBefore)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {0, 1});
+
+    cudf::table_view in{{ col1 }};
+    std::vector<cudf::order> order{ };
+    std::vector<cudf::null_order> null_precedence{ cudf::null_order::BEFORE };
+
+    auto actual = cudf::experimental::is_sorted(in, order, null_precedence);
+
+    EXPECT_EQ(true, actual);
+}
+
+TYPED_TEST(IsSortedNumeric, NullsBeforeFalse)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {1, 0});
+
+    cudf::table_view in{{ col1 }};
+    std::vector<cudf::order> order{ };
+    std::vector<cudf::null_order> null_precedence{ cudf::null_order::BEFORE };
+
+    auto actual = cudf::experimental::is_sorted(in, order, null_precedence);
+
+    EXPECT_EQ(false, actual);
+}
+
+TYPED_TEST(IsSortedNumeric, OrderArgsTooFew)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {1, 0});
+    fixed_width_column_wrapper<T> col2({0, 0}, {1, 0});
+
+    cudf::table_view in{{ col1, col2 }};
+    std::vector<cudf::order> order{ cudf::order::ASCENDING };
+    std::vector<cudf::null_order> null_precedence{ };
+
+    EXPECT_THROW(cudf::experimental::is_sorted(in, order, null_precedence),
+                 cudf::logic_error);
+}
+
+TYPED_TEST(IsSortedNumeric, OrderArgsTooMany)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {1, 0});
+
+    cudf::table_view in{{ col1 }};
+    std::vector<cudf::order> order{ cudf::order::ASCENDING,
+                                    cudf::order::ASCENDING };
+    std::vector<cudf::null_order> null_precedence{ };
+
+    EXPECT_THROW(cudf::experimental::is_sorted(in, order, null_precedence),
+                 cudf::logic_error);
+}
+
+TYPED_TEST(IsSortedNumeric, NullOrderArgsTooFew)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {1, 0});
+    fixed_width_column_wrapper<T> col2({0, 0}, {1, 0});
+
+    cudf::table_view in{{ col1, col2 }};
+    std::vector<cudf::order> order{ };
+    std::vector<cudf::null_order> null_precedence{ cudf::null_order::BEFORE };
+
+    EXPECT_THROW(cudf::experimental::is_sorted(in, order, null_precedence),
+                 cudf::logic_error);
+}
+
+TYPED_TEST(IsSortedNumeric, NullOrderArgsTooMany)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> col1({0, 0}, {1, 0});
+
+    cudf::table_view in{{ col1 }};
+    std::vector<cudf::order> order{ };
+    std::vector<cudf::null_order> null_precedence{ cudf::null_order::BEFORE,
+                                                   cudf::null_order::BEFORE };
+
+    EXPECT_THROW(cudf::experimental::is_sorted(in, order, null_precedence),
+                 cudf::logic_error);
+}
+
+template <typename T>
+struct IsSortedNvstringCategory : public cudf::test::BaseFixture {};
```

Comment:
NVStringCategory doesn't exist anymore. However, could you add some tests that include string columns? You can use `strings_column_wrapper` to help with creating/initializing columns of strings.